tests/unit/src/test_deinterleave16.c

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020-2022 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include <stdarg.h>
+
+#include "unity_fixture.h"
+
+#include "mic_array/util.h"
+
+#define CHAN_COUNT    16
+
+TEST_GROUP_RUNNER(deinterleave16) {
+  RUN_TEST_CASE(deinterleave16, case0);
+  RUN_TEST_CASE(deinterleave16, case1);
+}
+
+TEST_GROUP(deinterleave16);
+TEST_SETUP(deinterleave16) {}
+TEST_TEAR_DOWN(deinterleave16) {}
+
+
+/*
+This test ensures that the PDM samples corresponding to channel K end up in
+the kth word after deinterleaving.
+*/
+TEST(deinterleave16, case0)
+{
+
+  for(int k = 0; k < 16; k++){
+    uint32_t vals[CHAN_COUNT] = {0};
+
+    // Set only those bits corresponding to channel k.
+    // after deinterleaving, this should mean each vals[n] == 0, except for
+    // vals[k] == 0xFFFFFFFF
+    for(int n = 0; n < CHAN_COUNT; n++)
+      vals[n] = 0x00010001 << k;
+    
+    // printf("\n\n");
```

Comment:
Please either remove these commented out lines or add a comment explaining why they have been left in.

Reply:
They are useful so added a comment